src/graphql/types/abstract/initiation-via.ts

```diff
@@ -0,0 +1,43 @@
+import { GT } from "@graphql/index"
+
+import WalletId from "../scalar/wallet-id"
+import Username from "../scalar/username"
+import LnPaymentRequest from "../scalar/ln-payment-request"
+import OnChainAddress from "../scalar/on-chain-address"
+
+const InitiationViaIntraLedger = new GT.Object({
+  name: "InitiationViaIntraLedger",
+  fields: () => ({
+    walletId: {
+      type: GT.NonNull(WalletId),
+    },
+    counterPartyUsername: {
+      type: Username,
+    },
+  }),
+})
+
+const InitiationViaLn = new GT.Object({
+  name: "InitiationViaIntraLn",
+  fields: () => ({
+    paymentRequest: {
+      type: GT.NonNull(LnPaymentRequest),
```

Comment:
sometime the amount is mandatory (amountless invoice), sometimes it's explicitly forbidden. how are we dealing with this here?

because amount is somehow related to the initiation type